builder/dockerfile/dispatchers.go

```diff
@@ -111,6 +114,9 @@ func dispatchAdd(d dispatchRequest, c *instructions.AddCommand) error {
 // Same as 'ADD' but without the tar and remote url handling.
 //
 func dispatchCopy(d dispatchRequest, c *instructions.CopyCommand) error {
+	if c.Chmod != "" {
+		return errors.New("COPY --chmod requires BuildKit. Refer to https://docs.docker.com/go/buildkit/ to learn how to build images with BuildKit enabled.")
+	}
```

Comment:
Perhaps all command should take the same approach; `COPY` and `ADD` have some additional properties to store the `Chmod`, `Chown` values, but other commands may not; perhaps using the `FlagsUsed` approach could make it more generic (we could make a `supportedOptions` variable, containing the supported options per command.